### ISSUE TITLE
Add room whitelist to configuration

### ISSUE THREAD

**Issue by jonaharagon** [matrix-org/matrix-public-archive#234]
I think it would be nice if self-hosting admins could specify a list of rooms which can be joined by this client.

The use-case I'm envisioning is organizations running this archive for their own rooms, but not wanting to index every room accessible by the archive bot account. Trying to open a non-whitelisted room could either redirect to another archive, like archive.matrix.org, or simply return an error.

**Comment by MadLittleMods** [comment]
> The use-case I'm envisioning is organizations running this archive for their own rooms, but not wanting to index every room accessible by the archive bot account. 

If the goal is to keep the room out of a Matrix Public Archive instance, then it is probably better served by whatever solution we come up in https://github.com/matrix-org/matrix-public-archive/issues/47

---

Only allowing a small portion of rooms to be viewable from your Matrix Public Archive instance does sound like something viable for people who want to make their community accessible without making it a general archive client for everything else.

Depending on the exact use cases, this might be better served by a homeserver allowlist in combination with https://github.com/matrix-org/matrix-public-archive/issues/47 to remove whatever rooms you don't want showing.

**Comment by jonaharagon** [comment]
> people who want to make their community accessible without making it a general archive client for everything else.

Right, that's exactly the use-case I'm thinking of. I don't think #47 is quite the same because that covers _room owners_ opting-out of the archive, as opposed to _public-archive admins_ opting-out of archiving a room. Maybe a homeserver allowlist makes more sense, but I don't know if people would want to limit every user on their homeserver, just the `@archive` bot account, and I don't know if per-user join limits are possible to do in synapse?

**Comment by jonaharagon** [comment]
I think that my reading comprehension skills were not on point last week, if you meant an allowlist that only allows certain room directories from being enumerated by the public archive, that _does_ make sense to me. (For some reason I thought you meant restricting which rooms the bot could join on the server side)

The only problem with a room directory allowlist would be having to check whether a room is listed in an allowed directory, because if Matrix HQ isn't listed in my homeserver's room directory, I don't want people to be able to manually visit the `/r/matrix:matrix.org` URL and make my bot join that room either. So a room allowlist may still make more sense from this perspective.
